gradle-plugins/src/main/java/com/linkedin/pegasus/gradle/PegasusPlugin.java

```diff
@@ -1390,6 +1445,40 @@ protected void configureRestModelGeneration(Project project, SourceSet sourceSet
     });
   }
 
+  protected void configurePegasusSchemaSnapshotGeneration(Project project, SourceSet sourceSet, boolean isExtensionSchema)
+  {
+    File schemaDir = isExtensionSchema? project.file(getExtensionSchemaPath(project, sourceSet))
+        : project.file(getDataSchemaPath(project, sourceSet));
+    if (isExtensionSchema && !SharedFileUtils.getSuffixedFiles(project, schemaDir, PDL_FILE_SUFFIX).isEmpty())
+    {
+      project.getLogger().debug("No extension schemas, skip generatePegasusExtensionSchemaSnapshot task");
+      return;
+    }
+    File publishablePegasusSchemaSnapshotDir = project.file(project.getBuildDir().getAbsolutePath()
+        + File.separatorChar + sourceSet.getName() + (isExtensionSchema ? PEGASUS_EXTENSION_SCHEMA_SNAPSHOT: PEGASUS_SCHEMA_SNAPSHOT));
+
+    Task generatePegasusSchemaSnapshot = generatePegasusSchemaSnapshot(project, sourceSet,
+        isExtensionSchema ? PEGASUS_EXTENSION_SCHEMA_SNAPSHOT: PEGASUS_SCHEMA_SNAPSHOT, schemaDir, publishablePegasusSchemaSnapshotDir);
+
+    Task checkSchemaSnapshot = project.getTasks().create(sourceSet.getTaskName("check",
+        isExtensionSchema ? PEGASUS_EXTENSION_SCHEMA_SNAPSHOT: PEGASUS_SCHEMA_SNAPSHOT),
+        CheckPegasusSnapshotTask.class, task ->
+        {
+          task.dependsOn(generatePegasusSchemaSnapshot);
+          // TODO: update CheckPegasusSnapshotTask
+        });
+
+    File pegasusSchemaSnapshotDir = new File(isExtensionSchema ? getPegasusExtensionSchemaSnapshotPath(project, sourceSet)
+        : getPegasusSchemaSnapshotPath(project, sourceSet));
+    pegasusSchemaSnapshotDir.mkdirs();
```

Comment:
The `into` method on the `Copy` task (and the `Sync` task) configures the destination directory property of the task, which is annotated with `@OutputDirectory`. This means that Gradle will [create the directory for us if it is missing](https://docs.gradle.org/current/userguide/more_about_tasks.html#sec:task_input_output_validation), and this call to `mkdirs()` can be removed.

Reply:
Good to know that, removed.